docs/sources/developers/plugins/migration-guide.md

```diff
@@ -66,6 +67,98 @@ This guide helps you identify the steps required to update a plugin from the Gra
 
 ## From version 9.3.x to 9.4.x
 
+### Supporting new navigation layout
+
+First, enable the `topnav` feature flag in `custom.ini` to check how your plugin renders in the new navigation layout:
+
+```ini
+[feature_toggles]
+enable = topnav
+```
+
+#### Migrate from `onNavChanged`
+
+If your plugin uses the `onNavChanged` callback to inform Grafana of its nav model & sub pages, you should see that this results in duplicated navigation elements. If you disable `topnav` it should look just as before.
+
+When `topnav` is enabled we need to update the plugin to take advantage of the new `PluginPage` component and not call `onNavChanged`. `onNavChanged` is now deprecated.
+
+#### Switch to `PluginPage` component
+
+Grafana now exposes a new `PluginPage` component from `@grafana/runtime` that hooks into the new navigation and page layouts and supports the old page layouts when the `topnav` feature is disabled. This new component will also handle rendering the section navigation. The section navigation can include other core sections and other plugins. To control what pages are displayed in the section navigation for a specific plugin, Grafana will use the pages added in `plugin.json` that have `addToNav` set to `true`.
+
+This component is very easy to use. Simply wrap it around your page content:
+
+```tsx
+import { PluginPage } from '@grafana/runtime';
+
+...
+
+return (
+  <PluginPage>
+    {your page content here}
+  </PluginPage>
+);
+```
+
+Grafana will look at the URL to know what plugin and page should be active in the section nav, so this only works for pages you have specified in `plugin.json`. `PluginPage` will then render a page header based on the page name specified in `plugin.json`.
+
+#### Using `PluginPage` for pages not defined in `plugin.json`
+
+The `PluginPage` component also exposes a `pageNav` property that is a `NavModelItem`. This `pageNav` property is useful for pages that are not defined in `plugin.json` (e.g. individual item pages). The `text` and `description` you specify in the `pageNav` model will be used to populate the breadcrumbs and page header.
+
+Example:
+
+```tsx
+const pageNav = {
+  text: 'Write errors cortex-prod-04',
+  description: 'Incident timeline and details'
+};
+
+return (
+  <PluginPage pageNav={pageNav}>
+    {your page content here}
+  </PluginPage>
+);
+```
+
+The way the active page is matched in the breadcrumbs and section nav relies on the page routes being hierarchical. If you have a list page and an item page, the item page needs to be a subroute of the list page and the list page url needs to be specified in your `plugin.json`. For example, you might have a list of users at `/users`. This means that the item page for a specific user needs to be at `/users/:id`. This may require some refactoring of your routes.
+
```

Comment:
Would like to add an example of how to build a page with tabs: 

```tsx
const pageNav = {
    text: 'My page',
    description: 'Incident timeline and details',
    url: '/a/myorgid-pluginname-app',
    children: [
      {
        url: '/a/myorgid-pluginname-app/tab1',
        text: 'Tab1',
        active: true,
      },
      {
        url: '/a/myorgid-pluginname-app/tab2',
        text: 'Tab1',
      },
    ],
  };

return (
  <PluginPage pageNav={pageNav}>
    {your page content here}
  </PluginPage>
);
```

Reply:
@torkelo i've added a **very** basic skeleton for tabs documentation... do you know if there's an internal plugin that uses tabs that i can refer to and check my understanding? 🤔